### PR TITLE
Use f-strings where possible

### PR DESCRIPTION
**Pull request by akx**
# Description

* Relevant motivation: f-strings are cleaner to read and generally more performant than concatenation or `.format`.
* Summary of the change: wherever possible, use of `.format` etc. is replaced with f-strings. Initial automated conversion using ikamensh/flynt followed by manual fixes.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have added a short summary of my change to the CHANGELOG.md
- [x] My code follows the style guidelines of this project
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: gradio/interface.py [hunk 1]
```diff
@@ -385,7 +385,7 @@ def clean_html(raw_html):
                 if len(self.output_components) == 1:
                     component.label = "output"
                 else:
-                    component.label = "output " + str(i)
+                    component.label = f"output {str(i)}"
```
Comment (abidlabs) [inline gradio/interface.py:388]:
```suggestion
                    component.label = f"output {i}"
```

File: gradio/utils.py [hunk 2]
```diff
@@ -907,7 +904,7 @@ def tex2svg(formula, *args):
     DPI = 300
     plt.rc("mathtext", fontset="cm")
     fig = plt.figure(figsize=(0.01, 0.01))
-    fig.text(0, 0, r"${}$".format(formula), fontsize=FONTSIZE)
+    fig.text(0, 0, f"${formula}$", fontsize=FONTSIZE)
```
Comment (abidlabs) [inline gradio/utils.py:907]:
```suggestion
    fig.text(0, 0, fr"${formula}$", fontsize=FONTSIZE)
```

**Comment by abidlabs** [comment]
Thanks so much for fixing these @akx! Just noticed a couple of issues, which I'll commit and then merge in this branch

File: gradio/utils.py (hunk 2 above)
Comment (akx) [reply gradio/utils.py]: It doesn't matter, since $ isn't an escape sequence...

**Comment by akx** [comment]
> Thanks so much for fixing these @akx! Just noticed a couple of issues, which I'll commit and then merge in this branch

You're welcome! Should probably enable Ruff UP032 to catch these in the future, but maybe that's after #3979 is merged (the full Ruff UP suite will also change type annotations to new-style ones, which should be fine since most of the codebase is `from __future__ import annotations` anyway... anyway, that's a separate future PR :) )

File: gradio/utils.py (hunk 2 above)
Comment (abidlabs) [reply gradio/utils.py]: I think you're right lol

**Pull request merged**